admin/includes/modules/copy_product_confirm.php

```diff
@@ -47,52 +41,33 @@
-                                                      products_priced_by_attribute, product_is_free, product_is_call, products_quantity_mixed,
-                                                      product_is_always_free_shipping, products_qty_box_status, products_quantity_order_max,
-                                                      products_sort_order, products_price_sorter, master_categories_id)
-                  VALUES ('" . zen_db_input($product->fields['products_type']) . "',
-                          '" . $products_quantity . "',
-                          '" . zen_db_input($product->fields['products_model']) . "',
-                          '" . zen_db_input($product->fields['products_image']) . "',
-                          '" . $products_price . "',
-                          '" . zen_db_input($product->fields['products_virtual']) . "',
-                          now(),
-                          " . (zen_not_null(zen_db_input($product->fields['products_date_available'])) ? "'" . zen_db_input($product->fields['products_date_available']) . "'" : 'null') . ",
-                          '" . $products_weight . "', '0',
-                          '" . (int)$product->fields['products_tax_class_id'] . "',
-                          '" . (int)$product->fields['manufacturers_id'] . "',
-                          '" . zen_db_input(($product->fields['products_quantity_order_min'] == 0 ? 1 : $product->fields['products_quantity_order_min'])) . "',
-                          '" . zen_db_input(($product->fields['products_quantity_order_units'] == 0 ? 1 : $product->fields['products_quantity_order_units'])) . "',
-                          '" . zen_db_input($product->fields['products_priced_by_attribute']) . "',
-                          '" . (int)$product->fields['product_is_free'] . "',
-                          '" . (int)$product->fields['product_is_call'] . "',
-                          '" . (int)$product->fields['products_quantity_mixed'] . "',
-                          '" . zen_db_input($product->fields['product_is_always_free_shipping']) . "',
-                          '" . zen_db_input($product->fields['products_qty_box_status']) . "',
-                          '" . zen_db_input($product->fields['products_quantity_order_max']) . "',
-                          '" . zen_db_input($product->fields['products_sort_order']) . "',
-                          '" . zen_db_input($product->fields['products_price_sorter']) . "',
-                          '" . zen_db_input($categories_id) . "')");
+        $sql_data_array = array(); 
+        foreach ($product->fields as $key => $value) {
+          switch ($key) {
+          case products_id: 
+          case products_last_modified: 
+          case products_date_added: 
+          case products_date_available: 
+            continue 2;
+          }
+          $sql_data_array[$key] = $value; 
```

Comment:
NOOOOOOO! This would create a huge security hole.

`$value` here needs to be `zen_db_input($value)`

Reply:
Fixed -but this data is already in the database - it's not new user input.

Reply:
> Fixed -but this data is already in the database - it's not new user input.

Good point. I'd forgotten this was for "copy", and not collect/preview like other current open PRs are.

Reply:
It's not a bad change just in case bad data snuck in many moons ago.  Also it's good in case folks copy and paste this naively.